src/databricks/labs/remorph/transpiler/dialects_config.py

```diff
@@ -0,0 +1,27 @@
+from sqlglot.dialects.dialect import Dialect, Dialects
+
+from databricks.labs.remorph.snow.databricks import Databricks
+from databricks.labs.remorph.snow.snowflake import Snow
+from databricks.labs.remorph.snow.databricks_preview import DatabricksPreivew
+
+SQLGLOT_DIALECTS = {
+    "bigquery": Dialects.BIGQUERY,
+    "databricks": Databricks,
+    "databricks_preview": DatabricksPreivew,
+    "drill": Dialects.DRILL,
+    "mssql": Dialects.TSQL,
+    "netezza": Dialects.POSTGRES,
+    "oracle": Dialects.ORACLE,
+    "postgresql": Dialects.POSTGRES,
+    "presto": Dialects.PRESTO,
+    "redshift": Dialects.REDSHIFT,
+    "snowflake": Snow,
+    "sqlite": Dialects.SQLITE,
+    "teradata": Dialects.TERADATA,
+    "trino": Dialects.TRINO,
+    "vertica": Dialects.POSTGRES,
+}
+
+
+def get_dialect(engine: str) -> Dialect:
```

Comment:
make this entire file a private method of a class, that creates config. otherwise people would see this method easily publicly accessible and they start using it and we're back to square 1.

Reply:
Done